### ISSUE TITLE
adding Adan optimizer 

### ISSUE THREAD

**Issue by thegodone** [aimat-lab/gcnn_keras#75]
This very recent paper shows high potential but code is only available in pytorch: https://arxiv.org/abs/2208.06677

https://github.com/sail-sg/Adan/blob/main/adan.py

Can you add it ?

**Comment by PatReis** [comment]
I have added an implementation of adan in `kgcnn.training.optimizer` and I currently testing it in [tutorial_optimizer.ipynb](https://github.com/aimat-lab/gcnn_keras/blob/master/notebooks/tutorial_hyper_optuna.ipynb)

I am just a little uncertain whether for the first step the previous gradient should be 0 or just ignored via grad_0 - grad_-1 = 0.

**Comment by thegodone** [comment]
![image](https://user-images.githubusercontent.com/1186658/189181786-11407d24-bdeb-4db4-92e9-6ee4064b0556.png)

**Comment by thegodone** [comment]
did it works ?

**Comment by PatReis** [comment]
I think in the code there are small modifications to the algorithm presented above. 
But regarding the initialization, I think it rarely matters. Tensorflow does not has this restart condition and also initialization for step 0/-1 is a bit more tricky. But yes it works nicely either way. The difference to Adam is small.

![grafik](https://user-images.githubusercontent.com/64005383/189349671-f74c1022-b55f-47e6-8916-e75d2633ac7e.png)

I could post the implementation on keras-team github page ans ask how to handle start/restart initialization that already require first graodient.

**Comment by PatReis** [comment]
Can I close this or do you want me to as on keras-team repo?

**Comment by thegodone** [comment]
Please close but send to keras team too.


Envoyé de mon iPhone
